shared/src/store/action-reducers/utils/completely-load-vehicle.ts

```diff
@@ -0,0 +1,66 @@
+import type { Vehicle } from '../../../models';
+import {
+    isInVehicle,
+    isNotInTransfer,
+    VehiclePosition,
+} from '../../../models/utils';
+import { changePosition } from '../../../models/utils/position/position-helpers-mutable';
+import type { ExerciseState } from '../../../state';
+import type { Mutable } from '../../../utils';
+import { getElement } from './get-element';
+
+/**
+ * Checks whether all materials and personnel belonging to a vehicle are in it.
+ *
+ * @param draftState The state to operate in
+ * @param vehicle The vehicle to check
+ * @returns `true` if everything is in the vehicle, `false` otherwise
+ */
+export function isCompletelyLoaded(
+    draftState: Mutable<ExerciseState>,
+    vehicle: Mutable<Vehicle>
+) {
+    const materialsLoaded = Object.keys(vehicle.materialIds).map((materialId) =>
+        isInVehicle(getElement(draftState, 'material', materialId))
+    );
+
+    const personnelLoaded = Object.keys(vehicle.personnelIds).map(
+        (personnelId) =>
+            isInVehicle(getElement(draftState, 'personnel', personnelId))
+    );
+
+    return (
+        materialsLoaded.every((loaded) => loaded) &&
+        personnelLoaded.every((loaded) => loaded)
+    );
+}
```

Comment:
This assumes that if a thing is in A vehicle, the vehicle is loaded. This is currently correct, since material and personnel are strictly bound to their vehicle. But I think the usage of Vehicle.isInVehicle would be cleaner here since it additionally checks if the things are inside the correct vehicle

Reply:
Arguing this way, we should mark `isInVehicle` as deprecated, since it should never be used for any purpose than as perquisite for `currentVehicleIdOf` (as `Vehicle.isInVehicle`) does.

Changes the code in a way that personnel may enter multiple or any vehicle(-s) would lead to a big bunch of changes anyway while sticking to the generic `isInVehicle` here would at least keep the changes introduced by this PR small.

Reply:
Okay, you've got a point.

I've also changed the occurrences in `vehicle-popup.component.ts` to be consistent across this PR.